[Liu Yijun] iP

L1uY1jun opened the pull request:
## DukeGFMD

> You can find more information on GFMD [here](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax)

- This is a bullet list
- They are created with "*" or "-"

1. This is a numbered list
2. Each line is preceded with a number

``` java
// This is a fenced code block with comments.
public class Main {
    public static void main(String[] args) {
        System.out.println("Duke is cool!");
    }
}
```

This is a task list
- [ ] It are made with a hyphen, followed with a space then [ ].
- [x] It can be marked with a [x] too!

You can even mark `inline code` with backticks!

GFMD is useful! 💯 You can even do text formatting like so: ~~strikethrough~~, **bold** and *italics*.

Commit:
Add Gradle support

L1uY1jun committed:
Added Level 1. Greet, Echo, Exit.

L1uY1jun committed:
Modified Level 2. Add, List.

L1uY1jun committed:
Completed Level 3. Mark as Done, modified checkTask function

L1uY1jun committed:
Added Task class

L1uY1jun committed:
Implemented classes that inherit from Task class

L1uY1jun committed:
Completed Level 4. ToDos, Events, Deadlines

L1uY1jun committed:
Modified minor bugs

L1uY1jun committed:
Completed A-TextUiTesting

L1uY1jun committed:
Added new line to end of txt

L1uY1jun committed:
Defined a class DukeException to represent exceptions specific to Duke.

L1uY1jun committed:
Completed Level 5. Handle Errors.

L1uY1jun committed:
Completed Level 6. Delete.

L1uY1jun committed:
Duke class: change implementation of processCommand() method

L1uY1jun committed:
Task Class: Enable Constructor in subclasses to state isDone boolean when creating object

L1uY1jun committed:
Add Level 7. Save

L1uY1jun committed:
Add file /data/duke.txt and minor bug fix

L1uY1jun committed:
Add Level 8. Dates and Times

L1uY1jun committed:
Merge branch 'branch-Level-7'

L1uY1jun committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

L1uY1jun committed:
Add A-MoreOOP
Add A-Packages

L1uY1jun committed:
Add newline to end of files and minor change to string representation

L1uY1jun committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

L1uY1jun committed:
Add A-Gradle

L1uY1jun committed:
Add JUnit tests and ammend minor string representation error

L1uY1jun committed:
Modify string representation of DukeException in some code to remove special character

L1uY1jun committed:
Add A-Javadoc

L1uY1jun committed:
Add A-CodingStandard

L1uY1jun committed:
Add Javadoc for Duke class and modified some warnings

L1uY1jun committed:
Add Level 9. Find

L1uY1jun committed:
Merge branch 'branch-A-JavaDoc'

L1uY1jun committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/tools/Storage.java
#	src/main/java/duke/tools/TaskList.java

L1uY1jun committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/tasks/Task.java
#	src/main/java/duke/tools/Ui.java

L1uY1jun committed:
Add Javadocs and minor changes

L1uY1jun committed:
Add utility and more interactions from bot

L1uY1jun committed:
Add A-CheckStyle

L1uY1jun committed:
Complete Level-10, edit and enhance GUI from sample.

L1uY1jun committed:
Merge branch 'branch-Level-10'

charleslimjh commented on src/main/java/duke/gui/DialogBox.java:
```diff
@@ -0,0 +1,94 @@
+    /**
+     * Add a text bubble design for user text.
+     */
+    public void addUserTextDesign() {
+        this.dialog.setStyle("-fx-background-color: #0096FF;"
+                + "-fx-text-fill: white;"
+                + "-fx-min-width: 200px;"
+                + "-fx-wrap-text: true;"
+                + "-fx-font-size: 14;"
+                + "-fx-alignment: baseline-left;"
+                + "-fx-border-width: 0;"
+                + "-fx-background-radius: 15.0;"
+                + "-fx-padding: 15.0");
+    }
+
+    /**
+     * Add a text bubble design for bot text.
+     */
+    public void addBotTextDesign() {
+        this.dialog.setStyle("-fx-background-color: #DCDCDC;"
+                + "-fx-text-fill: black;"
+                + "-fx-min-width: 200px;"
+                + "-fx-wrap-text: true;"
+                + "-fx-font-size: 14;"
+                + "-fx-alignment: baseline-left;"
+                + "-fx-border-width: 0;"
+                + "-fx-background-radius: 15.0;"
+                + "-fx-padding: 15.0");
+    }
```
Really readable code here thanks to your line breaks. Great! 👍

charleslimjh reviewed (commented):
Your code looks really really good. 👍

Most public functions have well-written Javadocs that adhere to the coding standards. Code is quite readable as well and variables/functions are appropriately named. Keep it up!

highorbit25 commented on src/main/java/duke/gui/DialogBox.java:
```diff
@@ -0,0 +1,94 @@
+        getChildren().setAll(tmp);
+        setAlignment(Pos.TOP_LEFT);
+    }
+
```
Minor mistake of missing out Javadocs here

highorbit25 commented on src/main/java/duke/gui/DialogBox.java:
```diff
@@ -0,0 +1,94 @@
+        dialogBox.addUserTextDesign();
+        return dialogBox;
+    }
+
```
Minor mistake of missing out Javadocs here

highorbit25 commented on src/main/java/duke/tools/Storage.java:
```diff
@@ -0,0 +1,156 @@
+                    + "Tasks might not be stored.");
+        }
+    }
+}
```
Code is well structured and detailed Javadocs provided!

highorbit25 commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,14 @@
+list
+mark 1
+list
+bye
```
Overall very readable code. Nothing much to nitpick on for coding standard. Well done!

L1uY1jun committed:
Edit Javadoc of some methods

Some Javadoc were that were written for old method body, amended them
for the new method body and functionality.

L1uY1jun committed:
Add assertions to program

In the program, there were no assertion statements.

In order to determine whether our software is functioning with our
intentional suppositions, assertion statements are crucial.

Lets add some assertion statements to ensure our software functions
properly.

L1uY1jun committed:
Avoid magic literals and abstract out repeated strings

The code does not explain the meaning of the magic literals. Using a
named constant makes the code easier to understand because the name
tells us more about the meaning.

Repeated strings can be abstracted out for better readability.

Lets remove any magic literals and abstract out repeated strings for
better code quality.

L1uY1jun committed:
Avoid deep nesting and make happy path prominent

Deep nesting was found in Storage.java, part of the code can be
abstracted out to make the execution path clear and prominent.

Lets abstract out the part where data string is read into tasks for
better readability.

L1uY1jun committed:
Add minor changes on string format

L1uY1jun committed:
Use streams to manage task listing

A for loop was used to iterate through the list. This can have out of
bounds bug when we try to operate on tasks through the for loop.

Lets use streams instead which helps prevent bugs and increase
readability by making execution path clear and prominent.

L1uY1jun committed:
Merge pull request #1 from L1uY1jun/branch-A-CodeQuality

Branch-A-CodeQuality

L1uY1jun committed:
Merge pull request #2 from L1uY1jun/branch-A-Assertions

Add assertions to program

L1uY1jun committed:
Merge branch 'master' into branch-A-Streams

L1uY1jun committed:
Merge pull request #3 from L1uY1jun/branch-A-Streams

Use streams to manage task listing

L1uY1jun committed:
Add feature to update tasks

Previously there was no feature to update tasks.

To update a task, you need to delete the previous task and add a new
one, which can be troublesome.

Lets add an update command to update the task, be it a todo, event or
deadline directly.

L1uY1jun committed:
Improve GUI and improve code quality

GUI is unable to resize properly and there is no application icon.
There is also presence of magic literals and duplicate code.

The code does not explain the meaning of the magic literals. Using a
named constant makes the code easier to understand because the name
tells us more about the meaning and improve readability. Duplicate code
can also be abstracted out for better code quality.

Lets improve the GUI to make it more polished. Lets also abstract out
duplicate code and minimize use of magic literals by adding more named
constants.

L1uY1jun committed:
Add more interactions, edit naming of constants, and slight amendments
to OOP

L1uY1jun committed:
Bug fix and add a representative screenshot Ui.png

There was a bug where stored data was not read properly after previous
abstractions for deadline and event.

Lets fix the bug for proper execution. Lets also add a representative
screenshot of the application for setting up product website.

L1uY1jun committed:
Add user guide for AlphaBot

L1uY1jun committed:
Update User Guide and code reuse mentions

L1uY1jun committed:
Minor changes to user guide

L1uY1jun committed:
Update user guide

L1uY1jun committed:
Final update to user guide

L1uY1jun committed:
Optimize code and bug fix

L1uY1jun committed:
Update user guide

L1uY1jun committed:
Optimize GUI after smoke test result

Smoke test done by other peers have revealed a minor GUI problem that
the interface was allowed to resize beyond the minimum width.

Lets set a fixed minimum width to fix it.

L1uY1jun committed:
Update user guide documentation

L1uY1jun committed:
Fix coding standard issues for javadoc
